docs/tutorial/expressjs.rst

```diff
@@ -0,0 +1,430 @@
+need to load it into Docker:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:skopeo-copy]
+    :end-before: [docs:skopeo-copy-end]
+    :dedent: 2
+
+Check that the image was successfully loaded into Docker:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:docker-images]
+    :end-before: [docs:docker-images-end]
+    :dedent: 2
+
+The output should list the ExpressJS container image, along with its tag, ID and
+size:
+
+.. terminal::
+
+    REPOSITORY              TAG       IMAGE ID       CREATED       SIZE
+    expressjs-hello-world   0.1       30c7e5aed202   2 weeks ago   304MB
+
+.. note::
+    The size of the image reported by Docker is the uncompressed size which is
+    larger than the size of the compressed ``.rock`` file.
+
+Now we're finally ready to run the rock and test the containerised ExpressJS
+application:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:docker-run]
+    :end-before: [docs:docker-run-end]
+    :dedent: 2
+
+Use the same ``curl`` command as before to send a request to the ExpressJS
+application's root endpoint which is running inside the container:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:curl-expressjs-rock]
+    :end-before: [docs:curl-expressjs-rock-end]
+    :dedent: 2
+
+The ExpressJS application should again respond with the ``Welcome to Express``
+HTML.
+
+View the application logs
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When deploying the ExpressJS rock, we can always get the application logs via
+:ref:`pebble_explanation_page`:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: text
+    :start-after: [docs:get-logs]
+    :end-before: [docs:get-logs-end]
+    :dedent: 2
+
+As a result, Pebble will give us the logs for the
+``expressjs`` service running inside the container.
+We should expect to see something similar to this:
+
+.. terminal::
+
+    app@0.0.0 start
+    node ./bin/www
+    GET / 200 62.934 ms - 170
+
+We can also choose to follow the logs by using the ``-f`` option with the
+``pebble logs`` command above. To stop following the logs, press :kbd:`Ctrl` +
+:kbd:`C`.
+
+Cleanup
+~~~~~~~
+
+Now we have a fully functional rock for a ExpressJS application! This concludes
+the first part of this tutorial, so we'll stop the container and remove the
+respective image for now:
+
+.. literalinclude:: code/expressjs/task.yaml
+    :language: bash
+    :start-after: [docs:stop-docker]
+    :end-before: [docs:stop-docker-end]
+    :dedent: 2
+
+Chisel the rock
+===============
+
+This is an optional but recommended step, especially if we're looking to
+deploy the rock into a production environment. With :ref:`chisel_explanation`
+we can produce lean and production-ready rocks by getting rid of all the
+contents that are not needed for the ExpressJS application to run. This results
```

Comment:
```suggestion
we can produce lean and production-ready rocks by chipping away all the
contents that the application doesn't need to run. This will result
```